external/defects4c/scripts/checkout_bug.py

```
@@ -0,0 +1,148 @@
+def sanitize_project(project: str) -> str:
+    """Strip path components and restrict to safe chars; raise ValueError if invalid."""
+    name = os.path.basename(project)
+    if not name or name == "." or name == ".." or not SAFE_PROJECT_PATTERN.match(name):
+        raise ValueError(f"Invalid project name: {project!r}")
+    return name
+
+
+def run_git(cmd_args: list[str], cwd: str, timeout: int = 300) -> None:
+    """Run git with the given args; on timeout or non-zero exit, print message and exit 1."""
+    try:
+        subprocess.run(["git"] + cmd_args, check=True, cwd=cwd, timeout=timeout)
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)